test/integration/GuardedStream.test.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  RepresentationMetadata,
+  TypedRepresentationConverter,
+  readableToString,
+  ChainedConverter,
+  CONTENT_TYPE,
+  guardedStreamFrom,
+  RdfToQuadConverter,
+} from '../../src';
+import type {
+  Representation,
+  RepresentationConverterArgs,
+} from '../../src';
+
+class DummyConverter extends TypedRepresentationConverter {
+  public async getInputTypes(): Promise<Record<string, number>> {
+    return { '*/*': 1 };
+  }
+
+  public async getOutputTypes(): Promise<Record<string, number>> {
+    return { 'custom/type': 1 };
+  }
+
+  public async handle({ representation }: RepresentationConverterArgs): Promise<Representation> {
+    const data = guardedStreamFrom([ 'dummy' ]);
+    const metadata = new RepresentationMetadata(representation.metadata);
+
+    return { binary: true, data, metadata };
+  }
+}
+
+describe('A chained converter where data gets ignored', (): void => {
+  const identifier = { path: 'http://test.com/' };
+  const metadata = new RepresentationMetadata(identifier, { [CONTENT_TYPE]: 'text/turtle' });
```

Comment:
There's this new superhelpful contructor now 😉

```suggestion
  const metadata = new RepresentationMetadata(identifier, 'text/turtle');
```